lib/engine/game/g_1873/game.rb

```diff
@@ -3237,6 +3111,26 @@ def available_programmed_actions
             Action::ProgramIndependentMines,
           ]
         end
+
+        def aag_variant?
+          @aag_variant ||= @optional_rules&.include?(:aag_variant)
+        end
+
+        def optional_hexes
+          return self.class::HEXES unless aag_variant?
+
+          new_hexes = {}
+          HEXES.keys.each do |color|
+            new_map = self.class::HEXES[color].transform_keys do |coords|
+              coords - OPTION_REMOVE_HEXES
+            end
+            OPTION_ADD_HEXES.each { |coords, tile_str| new_map[coords] = tile_str } if color == :green
+
+            new_hexes[color] = new_map
+          end
+
+          new_hexes
+        end
```

Comment:
Why are you doing it this way? Did you see how I changed the map for 2P 18Mag?

Reply:
Rather than swapping out the standard F7 hex with the optional F7 hex, wouldn't it be easier to leave the F7 hex out at first and then just add one or the other? It's why I created game_hexes as a method rather than using the constant HEXES.